packages/type-utils/src/getConstrainedTypeAtLocation.ts

```diff
@@ -1,14 +1,20 @@
+import type {
+  ParserServicesWithTypeInformation,
+  TSESTree,
+} from '@typescript-eslint/typescript-estree';
 import type * as ts from 'typescript';
 
 /**
  * Resolves the given node's type. Will resolve to the type's generic constraint, if it has one.
  */
 export function getConstrainedTypeAtLocation(
```

Comment:
I wonder if this is common enough to add to our new wrapper?

Reply:
I think for now I'd rather steer clear of augmenting with our own APIs. I like sticking to 1:1 wrappers for simplicity's sake & having fewer API shapes to remember.

Maybe we'll eventually add caching for type functions such as `getBaseConstraintOfType`?